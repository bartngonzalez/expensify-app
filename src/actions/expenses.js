import uuid from 'uuid';

// ADD_EXPENSE
const addExpense = ({ description = '', note = '', amount = 0, createdAt = 0 } = {}) => {
	return {
		type: 'ADD_EXPENSE',
		expense: {
			id: uuid(),
			description: description,
			note: note,
			amount: amount,
			createdAt: createdAt
		}
	};
};

// REMOVE_EXPENSE
const removeExpense = ({ id } = {}) => {
	return {
		type: 'REMOVE_EXPENSE',
		id
	};
};

// EDIT_EXPENSE
const editExpense = (id, updates) => ({
	type: 'EDIT_EXPENSE',
	id,
	updates
});

export { addExpense, removeExpense, editExpense };
