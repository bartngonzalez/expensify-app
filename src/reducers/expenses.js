// Expenses Reducer

// Default State
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
	if (action.type == 'ADD_EXPENSE') {
		return [
			...state,
			action.expense
		];
	} else if (action.type == 'REMOVE_EXPENSE') {
		return state.filter(({ id }) => {
			return id !== action.id;
		})
	} else if (action.type == 'EDIT_EXPENSE') {
		return state.map((expense) => {
			if (expense.id == action.id) {
				return {
					...expense,
					...action.updates
				};
			} else {
				return expense;
			};
		});
	} else if (action.type == 'SET_EXPENSES') {
		return action.expenses;
	}
	return state;
};

export { expensesReducer };
