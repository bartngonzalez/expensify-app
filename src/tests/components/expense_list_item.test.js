import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/expense_list_item.js';
import { expenses } from '../fixtures/expenses.js';

test('should reneder ExpenseListItem', () => {
	const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
	expect(wrapper).toMatchSnapshot();
});
