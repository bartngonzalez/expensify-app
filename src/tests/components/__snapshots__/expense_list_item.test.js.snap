// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should reneder ExpenseListItem 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Car
    </h3>
  </Link>
  <p>
    3000
     - 
    0
  </p>
</div>
`;
